src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs

```diff
@@ -1413,7 +1415,8 @@ private void OnHeaderCore(HeaderType headerType, int? staticTableIndex, ReadOnly
         // https://tools.ietf.org/html/rfc7540#section-6.5.2
         // "The value is based on the uncompressed size of header fields, including the length of the name and value in octets plus an overhead of 32 octets for each header field.";
         _totalParsedHeaderSize += HeaderField.RfcOverhead + name.Length + value.Length;
```

Comment:
```suggestion
        _totalParsedHeaderSize += name.Length + value.Length;
```

Not new, but this RfcOverhead is for calculating the dynamic table size not total request header size. 32 bytes is a lot more overhead than a typical Http/1.1 request would have. No reason to be extra strict here either. We also enforce `MaxRequestHeaderCount` after all.

Reply:
The `SETTINGS_MAX_HEADER_LIST_SIZE` also mentions that this 32-byte overhead should be counted.

> SETTINGS_MAX_HEADER_LIST_SIZE (0x6):  This advisory setting informs a
      peer of the maximum size of header list that the sender is
      prepared to accept, in octets.  The value is based on the
      uncompressed size of header fields, including the length of the
      name and value in octets plus an overhead of 32 octets for each
      header field.